core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -1244,7 +1250,18 @@ private Response getObject(final String bucket,
           S3RangeSpec s3Range = S3RangeSpec.Factory.create(range);
           RangeFileInStream ris = RangeFileInStream.Factory.create(is, status.getLength(), s3Range);
 
-          Response.ResponseBuilder res = Response.ok(ris)
+          InputStream inputStream;
+          long rate =
+              (long) mSConf.getInt(PropertyKey.PROXY_S3_SINGLE_CONNECTION_READ_RATE_LIMIT_MB)
+                  * Constants.MB;
+          RateLimiter currentRateLimiter = S3RestUtils.createRateLimiter(rate).orElse(null);
+          if (currentRateLimiter == null && mGlobalRateLimiter == null) {
+            inputStream = ris;
+          } else {
+            inputStream = new RateLimitInputStream(ris, mGlobalRateLimiter, currentRateLimiter);
+          }
```

Comment:
we have a new architecture refactored s3 proxy api sits in S3ObjectTask#GetObjectTask#continueTask, would u be so kind to add the same ratelimiting change over there too? Thanks!

Reply:
We are using version 2.9.0. Will the new version be replaced with a new implementation?

Reply:
Set `alluxio.proxy.s3.v2.version.enabled=true`, and ignore the above reply.